cfme/tests/containers/test_provider_openscap_policy_attached.py

```diff
@@ -0,0 +1,87 @@
+
+AttributeToVerify = namedtuple('AttributeToVerify', ['table', 'attr', 'verifier'])
+
+TESTED_ATTRIBUTES__openscap = (
+    AttributeToVerify('configuration', 'OpenSCAP Results', bool),
+    AttributeToVerify('configuration', 'OpenSCAP HTML', lambda val: val == 'Available'),
+    AttributeToVerify('configuration', 'Last scan', dateparser.parse),
+    AttributeToVerify('compliance', 'Status', lambda val: val.lower() != 'never verified'),
+    AttributeToVerify('compliance', 'History', lambda val: val == 'Available')
+)
+
+
+@pytest.fixture(scope='function')
+def delete_all_container_tasks():
+    delete_all_tasks('AllTasks')
+
+
+@pytest.fixture(scope='function')
+def random_image_instance(appliance):
+    collection = appliance.collections.container_images
+    return random.sample(collection.all(), 1).pop()
+
+
+def get_table_attr(instance, table_name, attr):
+    # Trying to read the table <table_name> attribute <attr>
+    view = refresh_and_navigate(instance, 'Details')
+    table = getattr(view.entities, table_name, None)
+    if table:
+        return table.read().get(attr)
+
+
+@pytest.mark.polarion('10068')
+def test_check_compliance_provider_policy(provider, soft_assert,
+                                        random_image_instance):
+
+    # unassign OpenSCAP policy from chosen Image
+    random_image_instance.unassign_policy_profiles('OpenSCAP profile')
+
+    # assign policy from provider view
+    provider.assign_policy_profiles('OpenSCAP profile')
+
+    # Delete all old tasks in order to get fresh start
+    delete_all_container_tasks()
+
+    # Prform SSA Scan then check compliance with last know configuration
+    random_image_instance.perform_smartstate_analysis(wait_for_finish=True)
+
+    view = navigate_to(random_image_instance, 'Details')
+    for tbl, attr, verifier in TESTED_ATTRIBUTES__openscap:
+
+        table = getattr(view.entities, tbl)
+        table_data = {k.lower(): v for k, v in table.read().items()}
+
+        if not soft_assert(attr.lower() in table_data,
+                '{} table has missing attribute \'{}\''.format(tbl, attr)):
+            continue
+        provider.refresh_provider_relationships()
+        wait_for_retval = wait_for(lambda: get_table_attr(random_image_instance, tbl, attr),
+                                   message='Trying to get attribute "{}" of table "{}"'.format(
+                                       attr, tbl),
+                                   delay=5, num_sec=120, silent_failure=True)
+        if not wait_for_retval:
+            soft_assert(False, 'Could not get attribute "{}" for "{}" table.'
+                        .format(attr, tbl))
+            continue
+        value = wait_for_retval.out
+        soft_assert(verifier(value),
+            '{}.{} attribute has unexpected value ({})'.format(tbl, attr, value))
+
+    # unassign OpenSCAP policy from provider view
+    provider.unassign_policy_profiles('OpenSCAP profile')
```

Comment:
**REQUIRED**
This last line should in a fixture teardown.

Reply:
remove this last line. call fixture teardown openscap_assigned_rand_image